### ISSUE TITLE
Agent support

### ISSUE THREAD

**Issue by heepengpeng** [spring-projects/spring-ai#403]
**Expected Behavior**
Implement Agent like in Langchain with different method， such as CoT，ReACT

**Comment by markpollack** [comment]
If we do something like this it would be a post 1.0 feature.  FWIW, I found there to be varying definitions/implementations of 'agents', in particular I was able to reproduce langhchain/llamaindex demos of ReACT agent that did basic math (addition/subtraction/multiplcation) with basic functional calling behavior.

**Comment by kkolago** [comment]
I totally agree that implementing Agents would be very beneficial and would simplify writing more complex applications. I know that currently it is possible to model reasoning logic and interactions with external tools using function calling, advisors and evaluators, but the way you can model it in Langchain by simply defining graph of though process is much cleaner. Also maybe creating some interfaces for implementing custom agents would be nice I think (to make it more opinionated).

It would be nice to have something similar in Spring AI afer 1.0 GA (especially that we already have some tools that possibly might be leveraged for reasoning graph building like Spring Statemachine).

**Comment by lambochen** [comment]
> If we do something like this it would be a post 1.0 feature. FWIW, I found there to be varying definitions/implementations of 'agents', in particular I was able to reproduce langhchain/llamaindex demos of ReACT agent that did basic math (addition/subtraction/multiplcation) with basic functional calling behavior.

Have any relevant design documents or discussions?

**Comment by san-mateo-tech** [comment]
hello everyone, wanted to see if we have a design for a spring-agent, i did not find any and hence this post.  

i was thinking through and following is where i want to start thinking mathematically from the [universal machine intelligence equation](https://www.vetta.org/documents/legg-hutter-2007-universal-intelligence.pdf)  and abstract this upwards with appropriate dependency injection, with inspiration from the other coding agents while retaining unique value prop spring provides. 

<img width="553" height="76" alt="Image" src="https://github.com/user-attachments/assets/68cb5a84-f8c6-4513-8f4d-0fe32ed5d6bc" />

happy to receive feedbacks, guidances and directions.

**Comment by sdeleuze** [comment]
Currently planned for Spring AI 2.1, we will share more details and ask feedback later.

**Comment by johnyannj** [comment]
+1
